code/modules/surgery/organs/liver.dm

```diff
@@ -25,22 +25,17 @@
 			//slowly heal liver damage
 			damage = max(0, damage - 0.1)
 
-			if(filterToxins)
+			if(filterToxins && !owner.has_trait(TRAIT_TOXINLOVER))
 				//handle liver toxin filtration
-				var/toxamount
 				var/static/list/toxinstypecache = typecacheof(/datum/reagent/toxin)
 				for(var/I in C.reagents.reagent_list)
 					var/datum/reagent/pickedreagent = I
 					if(is_type_in_typecache(pickedreagent, toxinstypecache))
-						toxamount += C.reagents.get_reagent_amount(initial(pickedreagent.id))
-				if(toxamount <= toxTolerance && toxamount > 0)
-					for(var/I in C.reagents.reagent_list)
-						var/datum/reagent/pickedreagent = I
-						if(is_type_in_typecache(pickedreagent, toxinstypecache))
+						var/thisamount = C.reagents.get_reagent_amount(initial(pickedreagent.id))
+						if (thisamount <= toxTolerance && thisamount > 0)
 							C.reagents.remove_reagent(initial(pickedreagent.id), 1)
-				else if(toxamount > toxTolerance)
-					damage += toxamount*toxLethality
-
+						else
+							damage += min((thisamount <= toxTolerance ? 0 : (thisamount*toxLethality)),0)

```

Comment:
It's always going to not do any damage?

Reply:
Also please explain math like this with commenting so idiots like myself don't break anything in the future :)